### PR TITLE
Fix for json rpc compliancy

### PR DESCRIPTION
**Pull request by alsala**
Fixes for json-rpc:

https://horizenlabs.atlassian.net/browse/SDK-1660 (compliance with geth)
https://horizenlabs.atlassian.net/browse/SDK-1706 (compliance with standard)

The desired behavior described in SDK-1660 must be verified to be real compliance with geth and in case the fix should be reverted.

### CODE REVIEW

**Review by saratnt — APPROVED** [review]

File: sdk/src/main/scala/io/horizen/account/api/rpc/service/RpcProcessor.scala [hunk 1]
```diff
@@ -12,35 +12,49 @@ import scala.jdk.CollectionConverters.asScalaIteratorConverter
...
 
-case class RpcProcessor(val rpcHandler: RpcHandler) extends SparkzLogging {
+case class RpcProcessor(rpcHandler: RpcHandler) extends SparkzLogging {
+
+  def processEthRpc(body: JsonNode): (String, Boolean) = {
+
+    var jsonIsArray = false
 
-  def processEthRpc(body: JsonNode): String = {
     val requests = if (body.isArray && !body.isEmpty) {
```
[the remark this answers is not in the document]
Comment (alsala) [reply sdk/src/main/scala/io/horizen/account/api/rpc/service/RpcProcessor.scala:21]: Actually `RpcRequest` would not be called without that check, therefore I opted for handling all error conditions in that object.

[the remark this answers is not in the document]
Comment (saratnt) [reply sdk/src/main/scala/io/horizen/account/api/rpc/service/RpcProcessor.scala:21]: True, I confused the order of calls :D

File: qa/sc_evm_rpc_eth.py [hunk 2]
```diff
@@ -90,6 +128,128 @@ def run_test(self):
...
+        code, result = do_rpc_call(sc_node, payload)
+        check_result(code, result, expected_http_code=400, expected_string_in_result='Invalid request',
+                     expected_rpc_code=INVALID_REQUEST_CODE)
+
+        # batch request with just a failure, the expected output is a single error json
+        payload = json.dumps([1])
+        code, result = do_rpc_call(sc_node, payload)
+        assert_equal(400, code)
+        expected_result = {'error': {'code': -32600, 'message': 'Invalid request'}, 'jsonrpc': '2.0', 'id': None}
+        assert_equal(expected_result, result)
```
Comment (i-skrypnyk) [inline qa/sc_evm_rpc_eth.py:228]: Can the check_result as at 220 also be used at 226-228 ?

File: sdk/src/main/scala/io/horizen/account/api/http/route/AccountEthRpcRoute.scala [hunk 3]
```diff
@@ -56,7 +56,10 @@ case class AccountEthRpcRoute(
         _ =>
         {
           entity(as[JsonNode]) { body =>
-            SidechainApiResponse(rpcProcessor.processEthRpc(body));
+            {
+              val (json_resp, hasError) = rpcProcessor.processEthRpc(body)
+                SidechainApiResponse(json_resp, hasError)
```
Comment (i-skrypnyk) [inline sdk/src/main/scala/io/horizen/account/api/http/route/AccountEthRpcRoute.scala:61]: some formatting misaligned here

**Review by i-skrypnyk — APPROVED** [review]
LGTM!

**Pull request merged**